trailblazer/services/job_service/job_service.py

```diff
@@ -64,8 +71,24 @@ def get_analysis_status(self, analysis_id: int) -> TrailblazerStatus:
         if not analysis.jobs:
             raise NoJobsError(f"No jobs found for analysis {analysis_id}")
 
-        return get_status(analysis.jobs)
+        status: TrailblazerStatus = get_status(analysis.jobs)
+        if (
+            analysis.workflow_manager == WorkflowManager.TOWER
+            and status == TrailblazerStatus.COMPLETED
+        ):
+            status = TrailblazerStatus.QC
+        return status
```

Comment:
Where is this logic coming from?

Reply:
I dug it out from the old code: https://github.com/Clinical-Genomics/trailblazer/blob/daa140fca1743db625a7092fd772453c7e91031f/trailblazer/services/tower/tower_api_service.py#L62

All workflows that run in tower have a separate QC step which is why when the analysis completes the status is not set to `completed` but rather `qc`. I don't think it is a good design and it does not seem reasonable to assume that all workflows using tower will have a separate QC step.

I don't want to fix that in this PR.

Reply:
Agreed, that we shouldn't touch the logic. I just wanted to make sure that it wasn't something you cooked up yourself 😉